### PR TITLE
Shifted the AMQP IT test cluster from Debezium Kafka Cluster to Embedded Kafka Cluster

### PR DESCRIPTION
**Pull request by ShubhamRwt**
Signed-off-by: ShubhamRwt <srawat@redhat.com>

PR purpose :- 

This PR generally bumps the version of Kafka to 2.8 and shifts the test cluster for AMQP IT from Debezium Kafka Cluster to Embedded Kafka Cluster.

### CODE REVIEW

**Comment by Frawless** [comment]
Do we want this also for HTTP ITs? Or should we wait until @see-quick will figure out what we will do with `strimzi-test-container`?

File: src/test/java/io/strimzi/kafka/bridge/facades/KafkaFacade.java [hunk 1]
```diff
@@ -4,62 +4,65 @@
...
 
         Properties props = new Properties();
-        props.put("auto.create.topics.enable", "false");
-
-        kafkaCluster =
-            new KafkaCluster()
-                .usingDirectory(dataDir)
-                .withPorts(ZOOKEEPER_PORT, KAFKA_PORT)
-                .withKafkaConfiguration(props);
+        props.setProperty(KafkaConfig$.MODULE$.AutoCreateTopicsEnableProp(), "false");
```
Comment (scholzj) [inline src/test/java/io/strimzi/kafka/bridge/facades/KafkaFacade.java:38]: Wouldn't it be better to just use `"auto.create.topics.enable"`? It would be much more readable.

File: pom.xml [hunk 2]
```diff
@@ -287,12 +287,22 @@
 			<version>${hamcrest.version}</version>
 			<scope>test</scope>
 		</dependency>
+		<dependency>
+			<groupId>org.scala-lang</groupId>
+			<artifactId>scala-library</artifactId>
+			<version>2.13.6</version>
+		</dependency>
```
Comment (scholzj) [inline pom.xml:294]: Why do we need Scala?

File: pom.xml [hunk 3]
```diff
@@ -287,12 +287,22 @@
...
-			<artifactId>kafka_2.12</artifactId>
+			<artifactId>kafka_2.13</artifactId>
 			<version>${kafka.version}</version>
 			<scope>test</scope>
 		</dependency>
+		<dependency>
+		<groupId>org.apache.kafka</groupId>
+		<artifactId>kafka-streams</artifactId>
+		<version>${kafka.version}</version>
+		</dependency>
```
Comment (scholzj) [inline pom.xml:305]: Should this be scope => `test` only? Also, can you align it?

File: pom.xml [hunk 4]
```diff
@@ -306,6 +316,51 @@
 			<version>${debezium.version}</version>
```
Comment (scholzj) [inline pom.xml:316]: Do we still need the Debezium dependency if you replaced it with the `EmbeddedKafkaCluster`?

File: src/test/java/io/strimzi/kafka/bridge/amqp/AmqpBridgeIT.java [hunk 5]
```diff
@@ -216,10 +215,10 @@ void sendSimpleMessageToPartition(VertxTestContext context) throws InterruptedEx
                 KafkaConsumer<String, String> consumer = KafkaConsumer.create(this.vertx, config);
                 consumer.handler(record -> {
                     context.verify(() -> assertThat(record.value(), is(body)));
-                    // checking the right partition which should not be just the first one (0)
+//                     checking the right partition which should not be just the first one (0)
```
Comment (ppatierno) [inline src/test/java/io/strimzi/kafka/bridge/amqp/AmqpBridgeIT.java:218]: why this change?

File: src/test/java/io/strimzi/kafka/bridge/facades/KafkaFacade.java [hunk 6]
```diff
@@ -4,62 +4,65 @@
...
+        kafkaCluster = new EmbeddedKafkaCluster(1, props);
         return kafkaCluster;
     }
 
     public void createTopic(String topic, int partitions, int replicationFactor) {
-        kafkaCluster.createTopic(topic, partitions, replicationFactor);
+        try {
+            kafkaCluster.createTopic(topic, partitions, replicationFactor);
+        } catch (InterruptedException e) {
+            e.printStackTrace();
```
Comment (ppatierno) [inline src/test/java/io/strimzi/kafka/bridge/facades/KafkaFacade.java:49]: isn't it better using the LOGGER now that you have it?

Comment (ShubhamRwt) [reply src/test/java/io/strimzi/kafka/bridge/facades/KafkaFacade.java]: I did trie it the logger way I guess, we are using old Log4j 2.13. something which doesn't take object as an argument, while in the operator we use Log4j 2.14.1

Comment (ShubhamRwt) [reply src/test/java/io/strimzi/kafka/bridge/facades/KafkaFacade.java]: Should I update the Log4j dep too ?

**Comment by ShubhamRwt** [comment]
> Do we want this also for HTTP ITs? Or should we wait until @see-quick will figure out what we will do with `strimzi-test-container`?

@Frawless Hello, Iam not sure, I changed the cluster for AMQP because the debezium cluster was causing the AMQP tests fail on upgrading to Kafka 2.8.

**Review by scholzj — APPROVED** [review]

**Review by ppatierno — APPROVED** [review]

**Review by Frawless — APPROVED** [review]

**Comment by Frawless** [comment]
> > Do we want this also for HTTP ITs? Or should we wait until @see-quick will figure out what we will do with `strimzi-test-container`?
> 
> @Frawless Hello, Iam not sure, I changed the cluster for AMQP because the debezium cluster was causing the AMQP tests fail on upgrading to Kafka 2.8.

I see, maybe it's something we could discuss on the next community call.

**Comment by see-quick** [comment]
I think it would be better to be consistent in the case of this suite. The `strimzi-test-container` is ready to be used also in AMQP IT tests. I know I've had some problems with reworking AMQP tests in the past but today I played with it and it seems that `strimzi-test-container` works with AMQP tests also...so I wonder if this Embedded Kafka cluster is the ideal choice when we have `strimzi-test-container`. Anyway, I will talk also about this on the community call, which is tomorrow... :))

**Comment by ShubhamRwt** [comment]
@scholzj @see-quick Hello, Should I close this PR?

**Comment by see-quick** [comment]
Hi @ShubhamRwt, I have created PR (), which replaces Debezium Kafka by `strimzi-test-contaner`. So, I think you can close it.

**Comment by scholzj** [comment]
@see-quick @ShubhamRwt I don't know, we will need very soon a new release with Kafka 3.0.0. This one seems to be approved and tests passing. The other is draft with failing tests. So how much time does it need to get into a mergeable state?

**Comment by see-quick** [comment]
@scholzj It's ready now https://github.com/strimzi/strimzi-kafka-bridge/pull/542

**Comment by scholzj** [comment]
Replaced by #542
